tedana/decomposition/eigendecomp.py

```diff
@@ -266,13 +266,13 @@ def tedpca(catd, OCcatd, combmode, mask, t2s, t2sG,
 
     if len(ste) == 1 and ste[0] == -1:
         LGR.info('Computing PCA of optimally combined multi-echo data')
-        d = OCcatd[utils.make_min_mask(OCcatd[:, np.newaxis, :])][:, np.newaxis, :]
+        d = OCcatd[mask, :][:, np.newaxis, :]
```

Comment:
One more question: for the two other options, we explicitly cast to float64. Are we sure `OCcatd` is float64 already ?

Reply:
I don't see anywhere where the type for the optimally combined data is explicitly set. Honestly, I think casting to float64 is unnecessary for any of the data, but we can do it to the optimally combined data too just to be consistent.

Reply:
I'm fine with either path forward, I'm just pro-consistency unless otherwise specified :sweat_smile:

Reply:
Maybe we can open this as another issue, because I'm not sure how much it overlaps here, but I'm actually pretty sure your initial take (that casting to float64 is unnecessary) is correct. My reason for bringing it up now is because it's an easy chance to change it and I do think we're increasing memory usage with the re-casting....